dev

alekoChe opened the pull request.

alekoChe committed:
Попытка выполнить дз.
Не работает

alekoChe committed:
Убрал закомментированный код. Не работает.
Изменения в SimpleAuthService.java  и добавил
DataBase.java

efserg commented on src/main/java/db/DataBase.java:
```diff
@@ -1,34 +1,29 @@
             e.printStackTrace();
         } finally {
-            db.disconnect();  // когда приложение заканчивает работу
+            this.disconnect();  // когда приложение заканчивает работу
```
Почему вас удивляет, что приложение не работает?
Блок finally у вас выполнится не "когда приложение закончит работу", а сразу после создания объекта и соединение к базе закроется. А дальше вы пытаетесь выполнить запрос к закрытой базе

efserg reviewed (changes requested):
Пока оценку не ставлю, жду рабочий вариант

alekoChe committed:
Создал класс FileClient и в нем 3 метода:
1. createUserHistoryFileInForLoop(int i) // создает фаил (в котором должна храниться история клиента) в цикле for
2. writeInfoIntoFile(String info, String path) // Записываем инфу в файл
3. outputHistoryOfChat(String nick, int lengthOutList) //выводим историю чата для клиента 100 строк

В классе SimpleAuthService()
       создаем объект класса FileClient
    и применяем метод createUserHistoryFileInForLoop(i)

В классе ChatServer,
в методе sendMessageToClient:
- f.writeInfoIntoFile(mess, "allHistory.txt"); /// запись в файл allHistory
- f.writeInfoIntoFile(mess, pathSendersFile);  /// запись в файл отправителя
- f.writeInfoIntoFile(mess, pathAddresseesFile);  /// запись в файл адресата

В классе ChatClient:
  - controller.addMessage(f.outputHistoryOfChat(nick, 100));  // метод, который выводит историю чата

alekoChe committed:
добавил файлы с историей

alekoChe committed:
прикрутил db

alekoChe committed:
раскомментил createUserHistoryFileInForLoop(i)

alekoChe committed:
убрал некоторые комментарии

alekoChe committed:
изменения в файлах с историей

efserg commented on src/main/java/db/DataBase.java:
```diff
@@ -95,6 +126,39 @@ public void select(Integer id) throws SQLException {
+        }
+        return nick;
+    }
+    public String selectPass(Integer id) throws SQLException {
```
Зачем все эти методы? Что загрузить информацию по пользователям достаточно `select * from chatClients`, и дальше просто в цикле итерируетесь по resultSet'у, выбираю нужные поля. Все это показывалось на уроке, а тут наворочено не пойми чего

efserg commented on src/main/java/lesson3FileIO/FileClient.java:
```diff
@@ -0,0 +1,96 @@
+        Path path = Path.of( f.getPath() + "history_" + nick + ".txt");
+        List<String> list = null;
+        try {
+            list = Files.readAllLines(path);
```
Это будет катастрофой для большого файла с историей. Зачем читать весь файл, когда надо только 100 строк?

efserg commented on src/main/java/lesson3FileIO/FileClient.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    public static void main(String[] args) {
+//        FileClient f = new FileClient();
+//        File file = new File("C:\\Users\\aleko\\IdeaProject\\dbInChat", "myFile_01.txt");
+//        boolean exists = file.exists();
+//        System.out.println(exists);
+//
+//        File outputFile = new File("history_nick4.txt");
+//        String s = "Java hello!";
+//        f.writeInfoIntoFile(s, "history_nick4.txt");
+//
+//        //////////////////////////////////////////////
+//        String history = f.outputHistoryOfChat("nick0", 100);
+//        System.out.println(history);
```
Не надо оставлять неиспользуемый или закомментированный код в пулл-реквесте

alekoChe committed:
1 часть д.з. по lesson 4
создал: package threads --> public class WaitNotifyClass и в нем решение.

alekoChe committed:
2 часть д.з. Если честно не понял.
Вставил кэшдЭкзекьютор в конструктор клиентХендлера

alekoChe committed:
убрал закоммент. код

alekoChe committed:
убрал закоммент. код

efserg commented on src/main/java/server/ClientHandler.java:
```diff
@@ -24,16 +29,28 @@ public ClientHandler(Socket socket, ChatServer server) {
             this.server = server;
             this.in = new DataInputStream(socket.getInputStream());
             this.out = new DataOutputStream(socket.getOutputStream());
+            this.cachedService = Executors.newCachedThreadPool();
```
Здесь CachedThreadPool никогда не понадобится
